### PR TITLE
Update libraries.md for style issues

### PR DESCRIPTION
**Pull request by theletterf**
Updating the libraries page for style guide and language issues. Adding some comments where clarification might be required. This one really needed a cleanup.

### CODE REVIEW

File: content/en/docs/concepts/instrumentation/libraries.md [hunk 1]
```diff
@@ -210,41 +204,41 @@ through corresponding client implementation.
...
-- enrich RPC spans with absolutely essential library/service-specific
-  information not covered by auto-instrumentation
+- To support users without auto-instrumentation, which might not work in certain
+  environments or when users have concerns with monkey-patching.
+- To enable custom or legacy correlation and context propagation protocols with
+  underlying service.
+- Enrich RPC spans with essential library or service-specific information not
+  covered by auto-instrumentation.
 
-WARNING: Generic solution to avoid duplication is under construction 🚧.
```
Comment (theletterf) [inline content/en/docs/concepts/instrumentation/libraries.md:238]: @open-telemetry/docs-approvers Do we still need this? Not a big fan of "under construction" messages in docs. It's sort of implicit and can be verified by anyone interested.

File: content/en/docs/concepts/instrumentation/libraries.md [hunk 2]
```diff
@@ -291,25 +288,23 @@ try (Scope unused = span.makeCurrent()) {
...
-check out OpenTelemetry documentation in your language.
-
-In the case of a messaging system, you may receive more than one message at
+In the case of a messaging system, you might receive more than one message at
 once. Received messages become
-[_links_](/docs/languages/java/instrumentation/#create-spans-with-links) on the
+[links](/docs/languages/java/instrumentation/#create-spans-with-links) on the
 span you create. Refer to
 [messaging conventions](/docs/specs/semconv/messaging/messaging-spans/) for
-details (WARNING: messaging conventions are
```
Comment (theletterf) [inline content/en/docs/concepts/instrumentation/libraries.md:303]: @open-telemetry/docs-approvers Ditto on the need for this kind of messages.

File: content/en/docs/concepts/instrumentation/libraries.md (hunk 1 above)
Comment (cartermp) [reply content/en/docs/concepts/instrumentation/libraries.md:238]: I'm find with removing the more warning-y text.

**Review by cartermp — APPROVED** [review]

**Pull request merged**